sw/device/tests/aon_timer_wdog_lc_escalate_test.c

```diff
@@ -0,0 +1,242 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "sw/device/lib/base/freestanding/limits.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_alert_handler.h"
+#include "sw/device/lib/dif/dif_aon_timer.h"
+#include "sw/device/lib/dif/dif_pwrmgr.h"
+#include "sw/device/lib/dif/dif_rstmgr.h"
+#include "sw/device/lib/dif/dif_rv_plic.h"
+#include "sw/device/lib/dif/dif_rv_timer.h"
+#include "sw/device/lib/irq.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/alert_handler_testutils.h"
+#include "sw/device/lib/testing/aon_timer_testutils.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/rv_plic_testutils.h"
+#include "sw/device/lib/testing/test_framework/FreeRTOSConfig.h"
+#include "sw/device/lib/testing/test_framework/ottf.h"
+#include "sw/device/lib/testing/test_framework/test_status.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+const test_config_t kTestConfig;
+
+/**
+ * Program the alert handler to escalate on alerts upto phase 2 (i.e. reset) but
+ * the phase 1 (i.e. wipe secrets) should occur and last during the time the
+ * wdog is programed to bark.
+ */
+enum {
+  kWdogBarkMicros = 4 * 1000,          // 4 ms
+  kEscalationPhase0Micros = 3 * 1000,  // 3 ms
+  kEscalationPhase1Micros = 2 * 1000,  // 2 ms
+  kEscalationPhase2Micros = 500,       // 500 us
+};
+
+static_assert(kWdogBarkMicros > kEscalationPhase0Micros &&
+                  kWdogBarkMicros <
+                      (kEscalationPhase0Micros + kEscalationPhase1Micros),
+              "The wdog bark irq shall happens during the escalation phase 1");
```

Comment:
Sorry, just had a more in-depth look at the values here.

Note that the system reset occurs pretty instantaneously when entering Phase2, no matter how long `kEscalationPhase2Micros` is. So if we want to check that the watchdog gets disabled during escalation Phase1, and does not reset the system, it may make sense to configure the whatchdog so that both `Bark` and `Bite` happen during Phase1, before entering Phase2.